RHIDP-6567: HA support for AKS (#1100)
redhat-developer/red-hat-developers-documentation-rhdh: assemblies/assembly-configuring-high-availability.adoc

[id="{context}"]
= Configuring high availability in {product}

> High availability (HA) is a system design approach that ensures a service remains continuously accessible, even during failures of individual components, by eliminating single points of failure. It introduces redundancy and failover mechanisms to reduce downtime and maintain operational continuity.

Reviewer: ```suggestion
High availability (HA) is a system design approach that ensures a service remains continuously accessible, even during failures of individual components, by eliminating single points of failure. It introduces redundancy and failover mechanisms to minimize downtime and maintain operational continuity.
```

{product} supports HA deployments on {ocp-brand-name} and {aks-name}. The HA deployments enable more resilient and reliable service availability across supported environments.

> In a single instance deployment, a failure, whether due to software crashes, hardware issues, or other unexpected disruptions, would make the entire service unavailable, interrupting development workflows and access to key resources.

Reviewer: ```suggestion
In a single instance deployment, if a failure occurs, whether due to software crashes, hardware issues, or other unexpected disruptions, it would make the entire service unavailable, interrupting development workflows and access to key resources.
```

Author: Good catch. Thanks

With HA enabled, you can scale the number of backend replicas to introduce redundancy. This setup ensures that if one pod or component fails, others continue to serve requests without disruption. The built-in load balancer manages ingress traffic and distributes the load across the available pods. Meanwhile, the {product-very-short} backend manages concurrent requests and resolves resource-level conflicts effectively.

As an administrator, you can configure high availability by adjusting replica values in your configuration file:

* If you installed using the Operator, configure the replica values in your `{product-custom-resource-type}` custom resource.

> * If you used the Helm chart, set the replica values in the Helm configuration.

Reviewer: ```suggestion
* If you used the Helm chart, set the number of replicas in the Helm configuration.
```

Author: I think it’s best to leave it as is, since the changes pertain specifically to the replica values.
